torch/csrc/jit/codegen/cuda/scheduler/transpose.cpp

```diff
@@ -240,22 +273,35 @@ void maybeBuildVirtualInnerDims(
   //    both virtual innermost dim.
   // 2. The satisfied one did not merge in anything. For example,
   //    T0[I0{1024*1024}, I1{2}]
+  //    If this is the case, this means that we need to split the large
+  //    inner-most dimension to satisfy the small innermost dimension
   int64_t large_dim;
   int64_t split_factor;
+  bool split_inner_most;
   if (merged_size1 < params.tile_size1) {
     if (params.dims_merged_with_2.empty()) {
       // case 2
-      return;
+      split_inner_most = true;
+      large_dim = inner_most2;
+      split_factor = params.tile_size2;
```

Comment:
There is actually a bug. They shouldn't be identical, because case 1 has standalone innermost dim, while case 2's innermost dim will be merged with other dims. But they should be similar.